Add monitring overview to docs (#4338)
pipe-cd/pipecd: docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md

PipeCD comes with a monitoring system including Prometheus, Alertmanager, and Grafana.
This page walks you through how to set up and use them.

## Monitoring overview

![](/images/metrics-architecture.png)
<p style="text-align: center;">
Monitoring Architecture
</p>

> The piped agent collects its own monitoring information and periodically sends metrics to the Control plane, which publishes all of the resource usage and cluster information it collects and the metrics sent by the piped agent on its http server. When the PipeCD monitoring feature is turned on, Prometheus, Alertmanager and Grafana are deployed with the Control plane, and Prometheus retrieves metrics information from the Control plane's http server.

Reviewer: ```suggestion
The piped agent collects its metrics and periodically sends them to the Control plane. The Control plane then compacts its resource usage and cluster information with the metrics sent by the piped agent and re-publishes them via its admin server. When the PipeCD monitoring feature is turned on, Prometheus, Alertmanager, and Grafana are deployed with the Control plane, and Prometheus retrieves metrics information from the Control plane's admin server.
```

> The piped agent not only sends the collected monitoring metrics to the Control plane, but also provides them through its internal http server, so that the developer managing the piped agent can also get metrics directly from the piped agent and monitor them with their custom monitoring service.

Reviewer: ```suggestion
The piped agent not only sends the collected monitoring metrics to the Control plane, but also provides them through its admin server. The developers managing the piped agent can also use exposed metrics and monitor them with their custom monitoring service.
```

PipeCD comes with a monitoring system including Prometheus, Alertmanager, and Grafana.
This page walks you through how to set up and use them.

## Monitoring overview

![](/images/metrics-architecture.png)
<p style="text-align: center;">
Monitoring Architecture
</p>

> Both the Control plane and piped agent have their own "admin servers", which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.

Reviewer: ```suggestion
Both the Control plane and piped agent have their own "admin servers" (the default port number is 9085), which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```
